src/main/scala/com/gu/rcspollerlambda/services/DynamoDB.scala

```diff
@@ -1,23 +1,30 @@
 package com.gu.rcspollerlambda.services
 
 import com.gu.rcspollerlambda.config.Config._
-import com.gu.rcspollerlambda.models.{ DynamoReadError, LambdaError }
-import com.gu.scanamo.Scanamo
+import com.gu.rcspollerlambda.models.{ DynamoReadError, DynamoWriteError, LambdaError }
+import org.scanamo.{ Scanamo, Table }
+import org.scanamo.generic.auto._
+
+import scala.util.Try
 
 object DynamoDB extends Logging {
   case class LastId(id: String = "lastid", value: Long)
 
+  val scanamo = Scanamo(AWS.dynamoClient)
+
+  val table = Table[LastId](AWS.dynamoTableName)
+
   def getLastId: Either[LambdaError, Long] = {
     logger.info(s"Reading the last id from the ${AWS.dynamoTableName} table...")
-    Scanamo.scanWithLimit[LastId](AWS.dynamoClient)(AWS.dynamoTableName, 1).headOption
+    scanamo.exec(table.limit(1).scan()).headOption
```

Comment:
does this scanamo.exec also need to be in a Try block?

Reply:
I put the write into a Try mainly to attempt to keep the behaviour as similar as before. Although in reality, I think both Read and Write operations might have thrown exceptions before. I suppose it might be tidier to have both wrapped